solidity/dashboard/src/utils/token.utils.js

```diff
@@ -1,96 +1,192 @@
+   */
+  displayAmountWithSymbol = (amount) => {
+    return `${this.displayAmount(amount)} ${this.symbol}`
+  }
+
+  /**
+   * Returns a number with a metric suffix eg.:
+   * * 10000000 => { value: 10, suffix: 'M', formattedValue: '10M' }
+   * * 1000.4 => { value: 1.4, suffix: 'K', formattedValue: '1.4K'}
+   * * 10000.4 => { value: 10, suffix: 'K', formattedValue: '10K' }
+   * * 1 => { value: 1, suffix: '', formattedValue: '1' }
+   *
+   * @param {*} number
+   *
+   * @typedef {Object} NumebrWithSuffix
+   * @property {string} value - number in string
+   * @property {string} suffix - metric suffix
+   * @property {string} formattedValue - number with suffix
+   *
+   * @return {NumebrWithSuffix}
+   */
+  getNumberWithMetricSuffix = (number) => {
+    const bigNumber = new BigNumber(number)
+    let metric
+    for (let i = metrics.length - 1; i >= 0; i--) {
+      metric = metrics[i]
+      if (bigNumber.gte(metric.divisor)) {
+        break
+      }
+    }
+
+    const value = bigNumber.div(metric.divisor)
+
+    return {
+      value: value.toString(),
+      suffix: metric.symbol,
+      formattedValue: `${this.toFormat(
+        value,
+        metric.divisor === 1 ? this.decimalsToDisplay : 2
+      )}${metric.symbol}`,
     }
   }
 
-  let value = bigNumber.div(metric.divisor)
-  const beforeDecimal = value.toFraction(1)[0]
-  const precision = beforeDecimal.toString().length === 1 ? 1 : 0
-  value = bigNumber
-    .div(metric.divisor)
-    .toFormat(precision, BigNumber.ROUND_DOWN)
-
-  return {
-    value,
-    suffix: metric.symbol,
-    formattedValue: `${value}${metric.symbol}`,
+  /**
+   * Displays an amount with a metric suffix.
+   *
+   * @param {*} amount An amount to display.
+   *
+   * @return {string} Formatted amount with a metric suffix.
+   */
+  displayAmountWithMetricSuffix = (amount) => {
+    const result = this._displayAmount(amount, this.getNumberWithMetricSuffix)
+    return result?.formattedValue ? result.formattedValue : result
   }
-}
 
-export const displayAmountWithMetricSuffix = (amount) => {
-  return getNumberWithMetricSuffix(toTokenUnit(amount)).formattedValue
-}
+  _displayAmount = (amount, formattingFn = (amount) => amount) => {
+    if (!amount || isZero(amount)) {
+      return "0"
+    }
+
+    if (lt(amount, this.MIN_AMOUNT_TO_DISPLAY)) {
+      return `<${this.MIN_AMOUNT_IN_TOKEN_UNIT}`
+    }
 
-export const displayNumberWithMetricSuffix = (number) => {
-  return getNumberWithMetricSuffix(number).formattedValue
+    return formattingFn(this.toTokenUnit(amount))
+  }
 }
+
+export const KEEP = new Token(
+  "Keep Token",
+  18,
+  "KEEP",
+  "KEEP",
+  18,
+  Icons.KeepOutline,
+  0
+)
+
+export const ETH = new Token("Ether", 18, "ETH", "gwei", 14, Icons.ETH)
+export const TBTC = new Token("tBTC", 18, "TBTC", "tSats", 8, Icons.TBTC)
+export const LPToken = new Token(
+  "Liqudity Provider Token",
+  18,
+  "LP",
+  "LP",
+  18,
+  null,
+  4
```

Comment:
Last argument is not needed here because default value for `_decimalsToDisplay` is 4 so i think we can remove it.